### PR TITLE
Dockerization

### LINKED ISSUE #2: Dockerize the application
- [x] Add a Dockerfile
- [x] Add compose.yml
- [ ] All the code is reviewed and proved

### PR DESCRIPTION
**Pull request by annahramova97**
Dockerfile was created that allows deploying a container, however some setting for the project are missing (postgre initialization), that explains why the container is in Excited state, however, image is built successful.

image
image

This output is expected for now: 
image

In order to test you can try these commands:
1. docker build -t cocoa:1
2. docker images
3. docker run -d -it --name test cocoa:1
4. docker logs test

### CODE REVIEW

**Comment by AirinB** [comment]
@annahramova97 can we also set up (or install) PostgreSQL?

**Comment by annahramova97** [comment]
We tried to create new postgre container using this command:
 docker run --name testpostgre -e POSTGRES_PASSWORD=test -d postgres
The question is how to connect cocoa and postgresql now (docker-compose?).
Reading sensitive information it is possible using the secrets.

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+ENV PYTHONDONTWRITEBYTECODE=1
+
+# Turns off buffering for easier container logging
+ENV PYTHONUNBUFFERED=1
+
+RUN apt update && apt-get -y install sudo
+RUN apt-get install -y wget
+RUN wget --quiet -O - https://www.postgresql.org/media/keys/ACCC4CF8.asc
+RUN apt-get install -y postgresql
+
```
Comment (AirinB) [inline Dockerfile:14]: @annahramova97 is it possible to also set up the database in the docker file and when the users will set up the database for programming everything will already be set up?

**Review by AirinB — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 1 above)
Comment (annahramova97) [reply Dockerfile:14]: We can set it up using docker-compose, which will build both cocoa and postgresql as 2 services which are able to communicate

**Comment by annahramova97** [comment]
I added docker-compose.yml file that bounds db and app together. The values for the db initializiation are taken from the env.file. Cocoa service uses DB_URI from the env. file too. However the app still doesn't start as expected:
image

@AirinB Maybe you have any ideas? Could you check please env file to see if I set DB_URI as a correct pattern.

File: docker-compose.yml [hunk 2]
```diff
@@ -16,6 +16,7 @@ services:
       - ./.env
     environment:
       - DB_URI=${DB_URI}
```
Comment (annahramova97) [inline docker-compose.yml]: Not sure if it should be in db initialization or cocoa service itself?

File: Dockerfile (hunk 1 above)
Comment (AirinB) [reply Dockerfile:14]: Sounds good!

File: docker-compose.yml (hunk 2 above)
Comment (AirinB) [reply docker-compose.yml]: I think it should be the URL itself for the database which follows this pattern postgres://user:password@hostname/dbname
python -m cocoa.main

Comment (AirinB) [reply docker-compose.yml]: we could add here for the env var also the BOT_PREFIX: cocoapls. For the discord bot_token, I am not sure... Maybe we could use the GitHub secrets?

**Comment by annahramova97** [comment]
We have 2 instances of containers needed for this project, however, some problems are not eliminated, I guess we can leave it in this state for now since the current knowledge is not enough for now.
@AirinB @danilashtefan 
image

**Review by AirinB — APPROVED** [review]

**Comment by AirinB** [comment]
> We have 2 instances of containers needed for this project, however, some problems are not eliminated, I guess we can leave it in this state for now since the current knowledge is not enough for now.
> @AirinB @danilashtefan
> image

@annahramova97 yes, I also this so. We can leave it at this state, a big piece of work is already done anyway. Good job!

**Pull request merged**